src/metadata/ActionMetadata.ts

```diff
@@ -256,7 +256,7 @@ export class ActionMetadata {
     
     static appendBaseRouteToRegexpRoute(route: RegExp, baseRoute: string) {
         if (!baseRoute || baseRoute === "") return route;
-        const fullPath = baseRoute.replace("\/", "\\\\/") + route.toString().substr(1);
+        const fullPath = `^${baseRoute}${route.toString().substr(1)}?$`;
```

Comment:
Anyway, this is basically what [expressjs does under the hood](https://github.com/expressjs/express/blob/master/lib/router/layer.js#L16) but using [`path-to-regexp`](https://github.com/pillarjs/path-to-regexp) and the paths where not working properly due to a mismatch on the way the regex was being handled before. 

As `routing-controllers` is handling the route paths as strings directly, do you think it is better to add `path-to-regexp` as a dependency and use it instead of handling the regex manually?